src/test/java/part1/exercise/StreamsExercise2.java

```diff
@@ -24,25 +23,97 @@
 
     // TODO class PersonEmployerPair
 
+    private static class PersonEmployerPair {
+        String employer;
+        Person person;
+
+        public PersonEmployerPair(String employer, Person person) {
+            this.employer = employer;
+            this.person = person;
+        }
+
+        public String getEmployer() {
+            return employer;
+        }
+
+        public Person getPerson() {
+            return person;
+        }
+    }
+
+    private static Stream<PersonEmployerPair> employeeToPairs(Employee employee) {
+        return employee.getJobHistory().stream()
+                .map(JobHistoryEntry::getEmployer)
+                .map(p -> new PersonEmployerPair(p, employee.getPerson()));
+    }
+
+
     @Test
     public void employersStuffLists() {
-        Map<String, List<Person>> employersStuffLists = null;// TODO
-        throw new UnsupportedOperationException();
+        Stream<PersonEmployerPair> personEmployerPairStream = getEmployees().stream()
+                .flatMap(StreamsExercise2::employeeToPairs);
+        Map<String, List<Person>> employersStuffLists = personEmployerPairStream
+                .collect(Collectors.groupingBy(
+                        PersonEmployerPair::getEmployer,
+                        mapping(PersonEmployerPair::getPerson, toList())
+                ));
+
     }
 
     @Test
     public void indexByFirstEmployer() {
-        Map<String, List<Person>> employeesIndex = null;// TODO
-        throw new UnsupportedOperationException();
+        Map<String, List<Person>> employeesIndex = getEmployees().stream()
+                .map(e -> e.getJobHistory().stream().findFirst()
+                        .map(entry -> new PersonEmployerPair(entry.getEmployer(), e.getPerson())))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.groupingBy(
+                        PersonEmployerPair::getEmployer,
+                        mapping(PersonEmployerPair::getPerson, toList())
+                ));
     }
 
     @Test
     public void greatestExperiencePerEmployer() {
-        Map<String, Person> employeesIndex = null;// TODO
+        Map<String, Person> employeesIndex = getEmployees().stream()
+                .flatMap(employee -> employee.getJobHistory().stream()

```

Comment:
Fix indentation. Use method references instead of such big and complicated lambdas.